### PR TITLE
Add more bpftool tests

### PR DESCRIPTION
**Pull request by dthaler**
## Description

This PR is the next in the series of bpftool test PRs.  It adds more bpftool tests and fixes various issues they uncovered.

* Fix netsh delete program to delete all pinned paths and remove spurious "Program not found" output.
* Remove bpftool help text output that only applies to Linux at the moment
* Make bpftool tests work regardless of the actual program/map id value so the tests can be run multiple times
* Fix bugs in capture_helper that sometimes lost output

## Testing

This PR includes new tests.

## Documentation

No impact.

Resolves: #992

### CODE REVIEW

File: tests/bpftool_tests/bpftool_tests.cpp [hunk 1]
```diff
@@ -14,41 +14,20 @@
...
-    WaitForSingleObject(process_info.hProcess, INFINITE);
-
-    DWORD exit_code;
-    if (!GetExitCodeProcess(process_info.hProcess, &exit_code)) {
-        exit_code = GetLastError();
-    }
-
-    CloseHandle(process_info.hProcess);
-    CloseHandle(process_info.hThread);
+    *result = system(command_line);
```
Comment (Alan-Jowett) [inline tests/bpftool_tests/bpftool_tests.cpp:26]: Does capturing both stdout and stderr work? I thought this helper could only capture one or the other.

Comment (Alan-Jowett) [reply tests/bpftool_tests/bpftool_tests.cpp:26]: OK, I see it's fixed in this PR already.

**Review by Alan-Jowett — DISMISSED** [review]

**Comment by dthaler** [comment]
@Alan-Jowett rebased to resolve merge conflict so needs re-approval

**Review by Alan-Jowett — APPROVED** [review]

File: tests/bpftool_tests/bpftool_tests.cpp [hunk 2]
```diff
@@ -104,38 +78,53 @@ TEST_CASE("prog load map_in_map.o", "[!shouldfail][prog][load]")
...
     REQUIRE(result == 0);
 
-    output = run_command("netsh ebpf sh prog", &result);
-    REQUIRE(
-        output == "    ID  Pins  Links  Mode       Type           Name\n"
-                  "======  ====  =====  =========  =============  ====================\n"
-                  "196609     1      0  JIT        xdp            lookup\n\n");
+    output = run_command("bpftool prog show", &result);
     REQUIRE(result == 0);
+    std::string id = std::to_string(atoi(output.c_str()));
```
Comment (saxena-anurag) [inline tests/bpftool_tests/bpftool_tests.cpp:83]: I think we should do this to other netsh tests also. I was earlier trying to run the netsh tests for native code, but then was hitting this issue (the ID was changing).

**Review by saxena-anurag — APPROVED** [review]

**Pull request merged**